Add GUI interface for autogen with Gradio and GPT-Academic (#322)
microsoft/autogen: autogen/gradio_gui/README.md

# AutoGen's Gradio GUI

This is a GUI for AutoGen. It is written in Python and uses the [Gradio](https://gradio.app/) library.

## ⭐Installation

> 1. install AutoGen:

Reviewer: ```suggestion
1. Install AutoGen:
```

```bash
python -m pip install pyautogen
```

## ⭐Run

> 1. create `OAI_CONFIG_LIST`, write your OpenAI API key or azure API key. For example:

Reviewer: ```suggestion
1. Create a file named `OAI_CONFIG_LIST`, input your OpenAI or Azure OpenAI API key(s). For example:
```

```
[
    {
        "model": "gpt-3.5-turbo-16k",
        "api_key": "----------------------------------",

>         "api_base": "https://--------.openai.azure.com",

Reviewer: This needs to be updated when v0.2 is released.

Author: sorry,  v0.2 of what

Reviewer: v0.2 of pyautogen, no worries, just a reminder.

        "api_type": "azure",
        "api_version": "2023-07-01-preview",
        "deployment_id": "------"
    }
]

```

> 2. run following commands launch GUI:

Reviewer: ```suggestion
2. Run following commands to launch the GUI:
```

```bash
export OAI_CONFIG_LIST='/path/to/OAI_CONFIG_LIST'
export AUTOGEN_USE_DOCKER='False'
export WEB_PORT=12345
# for more environment variables options, please refer to the void terminal project

python -m autogen.launch_gui
```

> Note:

> > When you run `launch_gui`, the program will automatically install additional dependencies with `pip install --user ...` if they are not installed.

Reviewer: Would be nice to add a requirement.txt file so that users can install the dependencies by themselves.

Author: to install 
```
python -m pip install void-terminal>=0.0.9 https://github.com/binary-husky/gpt_academic/raw/master/docs/gradio-3.32.6-py3-none-any.whl
```

Reviewer: > to install
> 
> ```
> python -m pip install void-terminal>=0.0.9 https://github.com/binary-husky/gpt_academic/raw/master/docs/gradio-3.32.6-py3-none-any.whl
> ```

Add this into the Installation section?

> ## ⭐Customize

Reviewer: ```suggestion
## ⭐Customization
```

> 1. Create a python file `test.py` with templete below

Reviewer: Not necessary to be `test.py`, right?

> 2. Edit and add your own multiagent chat

Reviewer: ```suggestion
2. Edit and add your own group chat
```

3. Run it with `python test.py`

```python

from autogen.gradio_gui import init_config
from autogen.gradio_gui.utils.general import AutoGenGroupChat
from autogen.gradio_gui.plugin import autogen_terminal
from autogen.gradio_gui.gradio_service import main
import os
llm_config = init_config()

# <-------------------  define autogen agents (group chat)  ------------------->

> class AutoGenGroupChat(AutoGenGroupChat):

Reviewer: It would be nice to explain more details about the design and functionalities of the functions to be overridden.

Author: added in samples/apps/gradio_gui/general.py
